numerics/polynomial_in_monomial_basis.hpp

```diff
@@ -93,10 +93,8 @@ class PolynomialInMonomialBasis : public Polynomial<Value_, Argument_> {
   // powers of (argument - origin).
   constexpr PolynomialInMonomialBasis(Coefficients coefficients,
                                       Argument const& origin);
-  template<typename A = Argument,
-           typename = std::enable_if_t<is_vector_v<A>>>
-  explicit constexpr PolynomialInMonomialBasis(
-      Coefficients coefficients);
+  explicit constexpr PolynomialInMonomialBasis(Coefficients coefficients)
+    requires additive_group<Argument>;
```

Comment:
This is the weirdest constraint.  How is it sufficient for the argument to have an additive structure?  I mean, we'll compute `arg^N`.

Note that this is showing something problematic with the concepts: you require an addition, and if your argument doesn't have an addition the instantiation will fail.  But if you happen to use a multiplication, the concept will not prevent you from doing that, and you may end up with obscure errors.

Reply:
So far we have just been using concepts to replace enable_if, and we had been using that minimally to disable things that would not compile; `is_vector` was just a statement about the type of the difference, even weaker than additive_group.

We should move to concepts that more closely match the actual requirements, so that we get better errors; but we should do that from the bottom up, not from the middle in Polynomial.

The requirement for a multiplicative structure on Difference<Argument> should just be on Polynomial itself anyway, not here.